src/org/zalando/stups/mint/worker/job/run.clj

```diff
@@ -37,6 +41,7 @@
           (doseq [mint-app mint-apps]
             (let [app-id (:id mint-app)
                   kio-app (get kio-apps-by-id app-id)]
+              (when (or (not etcd-lock-url) (etcd/refresh-lock etcd-lock-url worker-id 600))
```

Comment:
I’d put this in the `when` clause above so we don’t even request apps from kio/mint when we’re not allowed to sync. Also a configurable etcd TTL would be nice imo.